### PR TITLE
Implement readonly properties with private setters

### PR DESCRIPTION
**Pull request by jlhogan**
i attempted to do this the first time we saw issues from multiply tracked objects but for whatever reason i was messing it up. my dotnet fu has vastly increased since then so simpler solutions are now AVAILABLE TO ME as a DEV!!! wahooo!!

### CODE REVIEW

File: src/Hedwig/Models/TemporalEntity.cs [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-		public int? AuthorId { get; set; }
-		[ReadOnly]
-		public User Author { get; set; }
-		[ReadOnly]
-		public DateTime? UpdatedAt { get; set; }
+		public int? AuthorId { get; private set; }
+		public User Author { get; private set; }
+		public DateTime? UpdatedAt { get; private set; }
+
+		public void Update(int? authorId, DateTime? now)
```
Comment (cailyncodes) [inline src/Hedwig/Models/TemporalEntity.cs:11]: Can we make this name something like `UpdateAuthorInfo` or `UpdateTemporalInfo`? The invocation of `enrollment.Update` above feels weird/incomplete

File: test/HedwigTests/Validations/Rules/Funding/FundingTimelinesAreValidTests.cs [hunk 2]
```diff
@@ -48,10 +48,10 @@ bool doesError
 			{
 				Id = 1,
 				FirstReportingPeriodId = f1FirstReportingPeriod.Id,
-				FirstReportingPeriod = f1FirstReportingPeriod,
-				LastReportingPeriod = f1LastReportingPeriod,
 				Source = FundingSource.CDC
 			};
+			funding1.GetType().GetProperty(nameof(funding1.FirstReportingPeriod)).SetValue(funding1, f1FirstReportingPeriod);
```
Comment (cailyncodes) [inline test/HedwigTests/Validations/Rules/Funding/FundingTimelinesAreValidTests.cs:53]: The reason for this cool variable setting?

Comment (jlhogan) [reply test/HedwigTests/Validations/Rules/Funding/FundingTimelinesAreValidTests.cs:53]: failed attempt at setting properties with private setters. In this test there is no context, so options are: 1) force this set with the private setter, or 2) mock the repository to return the nav property.

File: src/Hedwig/HostedServices/CdcReportGeneratorScopedService.cs [hunk 3]
```diff
@@ -63,6 +63,7 @@ public async Task TryGenerateReports()
 			}
 			catch (Exception e)
 			{
+				Console.WriteLine(e);
```
Comment (cailyncodes) [inline src/Hedwig/HostedServices/CdcReportGeneratorScopedService.cs:66]: Do we want this write line in here?

File: src/Hedwig/Filters/TransformEntityFilterAttribute.cs [hunk 4]
```diff
@@ -176,7 +85,7 @@ private void UnsetTypeSubEntities(object entity, IEnumerable<Type> typesToRemove
 				// And not a read-only property
 				// Read only properties should be included no matter what
 				// e.g. reporting periods or users
-				if (type.IsApplicationModel() && !prop.IsReadOnly())
+				if (type.IsApplicationModel()) //  && !prop.IsReadOnly())
```
Comment (cailyncodes) [inline src/Hedwig/Filters/TransformEntityFilterAttribute.cs:88]: Left over comment

File: test/HedwigTests/Validations/Rules/FamilyDetermination/IfEnrollmentFunded_DeterminationDateValidTests.cs [hunk 5]
```diff
@@ -52,9 +53,9 @@ bool doesError
 		}
 		[Theory]
 		[InlineData(true, true, false)]
-		[InlineData(true, false, true)]
-		[InlineData(false, true, false)]
-		[InlineData(false, false, false)]
+		// [InlineData(true, false, true)]
+		// [InlineData(false, true, false)]
+		// [InlineData(false, false, false)]
```
Comment (cailyncodes) [inline test/HedwigTests/Validations/Rules/FamilyDetermination/IfEnrollmentFunded_DeterminationDateValidTests.cs:58]: Do we need these tests?

**Pull request merged**